vpr/src/draw/draw.cpp

```diff
@@ -983,9 +984,9 @@ void init_draw_coords(float width_val) {
         draw_state->draw_rr_node = (t_draw_rr_node*)vtr::realloc(
             draw_state->draw_rr_node,
             (device_ctx.rr_nodes.size()) * sizeof(t_draw_rr_node));
-        for (size_t i = 0; i < device_ctx.rr_nodes.size(); i++) {
-            draw_state->draw_rr_node[i].color = DEFAULT_RR_NODE_COLOR;
-            draw_state->draw_rr_node[i].node_highlighted = false;
+        for (const RRNodeId& id : rr_graph.nodes()) {
+            draw_state->draw_rr_node[(size_t)id].color = DEFAULT_RR_NODE_COLOR;
```

Comment:
It would be great to leave a comment like

```
FIXME: the type case should be eliminated by making draw_rr_node adapt RRNodeId
```

Then other developers can later follow up.